src/encoded/static/components/item-pages/HiGlassViewConfigView.js

```diff
@@ -55,6 +57,79 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
         };
     }
 
+    static getGroupedLayouts = memoize(function (layouts) {
+        let groupedLayouts = null;
+        if (layouts.length > 1) {
+            groupedLayouts = _.groupBy(layouts, (it) => it.y);
+            const groupedLayoutKeys = _.keys(groupedLayouts);
+            const rowNames = groupedLayoutKeys.length === 3 ? ['top', 'middle', 'bottom'] : (groupedLayoutKeys.length === 2 ? ['top', 'bottom'] : null);
+            for (let i = 0; i < groupedLayoutKeys.length; i++) {
+                const rowItem = groupedLayouts[groupedLayoutKeys[i]];
+                const colNames = rowItem.length === 3 ? ['left', 'center', 'right'] : (rowItem.length === 2 ? ['left', 'right'] : null);
+                for (let j = 0; j < rowItem.length; j++) {
+                    rowItem[j].displayText = [
+                        groupedLayoutKeys.length > 3 ? (i + 1) : ((rowNames && rowNames[i]) || null),
+                        rowItem.length > 3 ? (j + 1) : ((colNames && colNames[j]) || null)
+                    ].filter(Boolean).join(' - ');
+                }
+            }
+        } else {
+            const data = {};
+            groupedLayouts = data[layouts[0].y.toString()] = [layouts];
+        }
+        return groupedLayouts;
+    }, function (A, B) {
+        const arrA = A[0];
+        const arrB = B[0];
+        if (arrA.length !== arrB.length) { return false; }
+        for (let i = 0; i < arrA.length; i++) {
+            if ((arrA[i].x !== arrB[i].x) || (arrA[i].y !== arrB[i].y)) { return false; }
+        }
+        return true;
+    });
+
+    static getTilesetUids(obj) {
```

Comment:
Would add a comment that `obj` is being modified in place here (obj.views = ... ) because it receives a fresh object each time the func is called.